File: Samples/PurchasePath/Source/Objective-C/MainController.m
#import "MainController.h"
#import <Button/Button.h>
#import "SampleExtension.h"

@interface MainController ()

@property (nonatomic, strong) NSURL *merchantURL;

@end

@implementation MainController


- (void)viewDidLoad {
    [super viewDidLoad];
    _merchantURL = [[NSURL alloc] initWithString:@"http://jet.com"];
    self.tableView.delegate = self;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    SampleExtension *extension = indexPath.row == 0 ? [SampleExtension new] : nil;
    [self fetchAppActionWithUrl:_merchantURL purchasePathExtension:extension];
}

- (void)fetchAppActionWithUrl:(NSURL *)url purchasePathExtension:(id <BTNPurchasePathExtension>)purchasePathExtension {
    Button.purchasePath.extension = purchasePathExtension;
    BTNPurchasePathRequest *request = [BTNPurchasePathRequest requestWithURL:url];
    [Button.purchasePath fetchWithRequest:request purchasePathHandler:^(BTNPurchasePath *purchasePath, NSError *error) {
        if (purchasePath) {
            [purchasePath start];
        }
    }];
}


@end

File: Samples/PurchasePath/Source/Swift/MainController.swift
import UIKit
import Button

class MainController: UITableViewController {
    
    /// Edit this URL to be any approved merchant in order to test the Checkout flow with or without the extension.
    private var yourExampleMerchantUrl = URL(string: "https://jet.com")!

    override func viewDidLoad() {
        super.viewDidLoad()
        tableView.delegate = self
    }
    

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        switch indexPath.row {
        case 0:
            fetchAppAction(using: yourExampleMerchantUrl, with: SampleExtension())
        case 1:
            fetchAppAction(using: yourExampleMerchantUrl, with: nil)
        default:
            ()
        }
    }
    
    
    private func fetchAppAction(using url: URL, with purchasePathExtension: PurchasePathExtension?) {
        Button.purchasePath.extension = purchasePathExtension;
        let request = PurchasePathRequest(url: url);
        Button.purchasePath.fetch(request: request) { appAction, error in
            appAction?.start();
        }
    }
}
